### PR TITLE
Implementation of tool lints.

### PR DESCRIPTION
**Pull request by flip1995**
Tracking issue: #44690

### CODE REVIEW

File: src/libsyntax/attr.rs [hunk 1]
```diff
@@ -308,6 +315,10 @@ impl MetaItem {
     pub fn is_meta_item_list(&self) -> bool {
         self.meta_item_list().is_some()
     }
+
+    pub fn is_scoped(&self) -> bool {
```
Comment (oli-obk) [inline src/libsyntax/attr.rs]: You could make this return an option of the first segment and then use that segment in the `is_known_lint_tool` function (and maybe to use a more fine-grained span for the diagnostic)

Comment (flip1995) [reply src/libsyntax/attr.rs]: Yeah good idea!

**Comment by oli-obk** [comment]
@bors r+

lgtm

**Pull request merged**